rskj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -767,6 +767,16 @@ private long computeDataCopyGas() {
         return calcMemGas(oldMemSize, newMemSize, copySize);
     }
 
+    private long computeMemoryCopyGas() {
```

Comment:
This is still only charging for memory expansion, but ignoring the copy cost. Please update following the specs

https://github.com/ethereum/EIPs/blob/master/EIPS/eip-5656.md#gas-costs

Reply:
@donequis , do you mean the constant MCopy cost (value of `g_verylow ` from the formula) ?

Reply:
No. There's a price pay for word copied.  `3 * words_copied`

Reply:
.. if so, it's defined in [here](https://github.com/rsksmart/rskj/pull/2711/files#diff-18fc2e78acdfcad02b2843ae8f0d34ba6d216f6a38641c4735db7a85e4e2c7c9R334), and used in [here](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/org/ethereum/vm/VM.java#L2083)

Reply:
So the formula has 3 components. First is memory expansion which is considered here. Second is the basic cost. Third part is the length copied. This las part is not considered in the cost.

Reply:
isn't that part in [here](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/org/ethereum/vm/VM.java#L162-L166)?

Reply:
Oh, you're right. My bad.

Reply:
Then we are good to go.

Reply:
no worries. It's good to validate assumptions and align on that